pkg/scale/bitvec.go

```diff
@@ -0,0 +1,91 @@
+// Copyright 2023 ChainSafe Systems (ON)
+// SPDX-License-Identifier: LGPL-3.0-only
+
+package scale
+
+const (
+	// maxLen equivalent of `ARCH32BIT_BITSLICE_MAX_BITS` in parity-scale-codec
+	maxLen = 268435455
+	// byteSize is the number of bits in a byte
+	byteSize = 8
+)
+
+// BitVec is the implementation of the bit vector
+type BitVec struct {
```

Comment:
I don't think this should belong in the `scale` package.  Given that we will probably need a `BitVec` at some other point, I think it makes sense to create a `bitvec` package to house this?

Reply:
In that case, shouldn't we do that for `uint128` and `VaryingDataType` as well?

Reply:
Yea.. I put those two in the `scale` package for other reasons:
- In rust `u128` is primitive type, and we don't have a native 128 bit unsigned integer implementation at this point.   I can argue the other side, where they should be their packages, but since the SCALE type system is essentially a subset of rust primitives, I thought it made sense to keep them in the same package.
- Similarly `VaryingDataType`/`enum` and `Result` are not supported in go natively, but are part of the supported types as defined in the [spec](https://spec.polkadot.network/id-cryptography-encoding#sect-scale-codec).

Looks like in the spec they do mention the use of `bitvec::BitVec` as [Hex encoding](https://spec.polkadot.network/id-cryptography-encoding#sect-scale-codec).  If we never use anything else from the `bitvec` other than `BitVec` I don't mind the go version of `BitVec` to live in `scale`.

Reply:
Makes sense. I had used the `pkg/scale` in one of my projects before and found it convenient just to import one package. 

Centrifuge mentions our package [here](https://github.com/centrifuge/go-substrate-rpc-client/tree/master/scale). I feel we should also consider the external usability.

Reply:
> Centrifuge mentions our package here and it looks like the other scale codec lib isn't being maintained. I feel we should also consider the external usability.

I think this was before I did the scale rewrite for our codebase.

Reply:
Given the incomplete nature of this PR does this need to be merged into `development` now? Maybe this can be included on the release branch for the parachain work? I'd rather not introduce code into `development` that isn't used.

Reply:
This is a blocker for https://github.com/ChainSafe/gossamer/pull/3241. Are we looking to merge all the parachain work to this release branch?

Reply:
Yes I think that makes sense.  We are doing the same for the GRANDPA work.

Reply:
Ok.